datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -2851,4 +2838,70 @@ mod tests {
             or(col("c2").lt(lit(3)), col("c2").gt(lit(4)))
         );
     }
+
+    #[test]
+    fn test_like_and_ilke() {
+        // test non-null values
+        let expr = Expr::Like(Like::new(
+            false,
+            Box::new(col("c1")),
+            Box::new(lit("%")),
+            None,
+        ));
```

Comment:
What do you think about using the expr_fn's style for creating `like` https://docs.rs/datafusion/latest/datafusion/prelude/enum.Expr.html#method.like

```suggestion
        let expr = like(col("c1"), lit("%"));
```

I think that would make it clearer what cases these tests were covering

Reply:
Yes, it is a better way. I forgot to use it 😄